src/WarcraftLegacies.Source/Quests/Warsong/QuestWarsongOutpost.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Warsong
+{
+  /// <summary>
+  /// The Warsong clan can acquire the Warsong Outpost in Uldum.
+  /// </summary>
+  public sealed class QuestWarsongOutpost : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+    private readonly ObjectiveAnyUnitInRect _enterWarsongOutpostRegion;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestWarsongOutpost"/>.
+    /// </summary>
+    public QuestWarsongOutpost() : base("Warsong Outpost in Uldum",
```

Comment:
This is deeply uninspired. The standard for quest names is for them to be a bit artful. E.g. Arthas becomes a Death Knight via "The Dark Champion" rather than "Death Knight Arthas".

Reply:
Updated to something I hope is more flavourful, especially with the suggestion you added :)